public/docs/ts/latest/tutorial/toh-pt6.jade

```diff
@@ -0,0 +1,252 @@
+      .file mock-heroes.ts
+    .file node_modules ...
+    .file typings ...      
+    .file index.html
+    .file package.json
+    .file styles.css
+    .file tsconfig.json
+    .file typings.json
+:marked
+  ### Keep the app transpiling and running
+  Open a terminal/console window and enter the following command to
+  start the TypeScript compiler, start the server, and watch for changes:
+
+code-example(format="." language="bash").
+  npm start
+
+:marked
+  The application runs and updates automatically as we continue to build the Tour of Heroes.
+
+.l-main-section
+:marked
+  ## Heroes and Http
+  
+  Let's look back at our `HeroService` implementation in the <a href="/docs/ts/latest/tutorial/toh-pt4.html">services</a> chapter. Specifically let's revisit the point we made about using promises when loading heroes. 
+  
+  Wrapping a simple constant in a promise might have seemed like overkill at the time, but we will show why this was an important first step towards moving to `Http`. 
+  
+  Let's get started by converting `getHeroes()` to use `Http`.
+  
++makeExample('toh-6/ts/app/hero.service.ts', 'get-heroes', 'app/hero.service.ts (getHeroes using Http)')(format=".")
+  
+:marked
+  ### Http Promise
+
+  In our previous implementation we created a promise ourselves and resolved it immediately. When using `Http` we no longer have to create the promise manually. Instead `Http` will create it for us, but we have to call `toPromise()` to get it.
+  
+  In the first `then` we convert the response to a json object by calling `res.json()`.
+  
+  Our api follows the convention of returning the result wrapped in a `data` property, but we don't want to expose this to the caller, so we flatten the response in the second `then` by returning `res.data`.   
+
+:marked
+  ### Error Handling
+  
+  At the end we are calling `catch` and passing in an error handler. This is an important step since it allows us to catch any error returned by the api. 
++makeExample('toh-6/ts/app/hero.service.ts', 'error-handler', 'app/hero.service.ts (Error handler)')(format=".")
+
+:marked
+  ### Promises are Promises
+  Even though we made significant internal changes to `getHeroes()`, the public signature did not change. Like before we are returning a promise, so the callers of `getHeroes()` will not need to be updated.  
+
+.l-sub-section
+  :marked
+    We want to point out that a promise is not the only way to process http calls. We will encounter other approaches such as Observables in subsequent chapters, but for the purposes of this article we are sticking to promises. 
+
+.l-main-section   
+:marked
+  ## Add, Edit, Delete
+  
+  Our stakeholders are incredibly pleased with the added flexibility from the api integration, but it doesn't stop there. Next we want to add the capability to add, edit and delete heroes.   
+  
+  In this section we will show how to implement http `post`, `put` and `delete` calls in `HeroService` to meet our new requirements. 
```

Comment:
I don't think we need to lead off with "In this section we will ...". We are in the section and we can just tell them what we are doing. This will provide a more active conversation instead of a passive sentence.

We'll complete our service by creating `post`, `put`, and `delete` http calls in our `HeroService` to meet our new requirements.